### DISCUSSION TITLE
Problem with Unit

### DISCUSSION THREAD

**Discussion by kamiro16** [YetiForceCompany/YetiForceCRM#15701] [How to] [answered]
Hello! 

In one of the calculation modules, I noticed that the unit is not completed automatically. I tried to remove or add the field, but the effect is the same. Anyone have an idea how to associate the field with the product?

[PL] 
Zauważyłem, że w jednym module z kalkulacjami pole jednostka nie jest uzupełniane automatycznie. 
Co może być powodem? 

Poniżej screen: 
<img width="1344" alt="Zrzut ekranu 2022-02-9 o 11 18 15" src="https://user-images.githubusercontent.com/65873917/153178720-17eca648-2b4b-4a7d-99b3-64b0d19d91f6.png">

**Comment by Cattrah** [comment, marked as answer]
I also noticed this problem when creating an advanced block on a custom module or an existing module that does not come pre-loaded with an advanced block. I tried to manually add a mapping to the _invmap _invfield _inventory tables for the module, but it does not autopopulate these fields. I am guessing it has something to do with autofill but I can't find the place on the front end or the back end to hook up this relationship.

**Reply by Cattrah** [reply]
Update: My issue was when I added the mapping to my u_yf_custommodule_invmap table, I did not capitalize Products as the module name. Once I did that, it started working fine. So for your issue, you probably just need to check the _invmap _invfield and _inventory tables of your module to make sure they have the correct mappings and then it should work, it seems that when you add these fields from the front end it doesn't always add the mapping correctly.
